src/ContentClient.ts

```diff
@@ -148,42 +154,50 @@ export class ContentClient implements ContentAPI {
         const queries = splitManyValuesIntoManyQueries(this.contentUrl, '/deployments', queryParams, reservedChars)
 
         // Perform the different queries
-        const results: AddedAudit[] = []
-        for (const query of queries) {
+        const foundIds: Set<EntityId> = new Set()
+        let exit = false
+        for (let i = 0; i < queries.length && !exit; i++) {
+            const query = queries[i]
             let offset = 0
             let keepRetrievingHistory = true
-            while (keepRetrievingHistory) {
+            while (keepRetrievingHistory && !exit) {
                 const url = query + (queryParams.size === 0 ? '?' : '&') + `offset=${offset}`
-                const partialHistory: PartialDeploymentHistory<AddedAudit> = await this.fetcher.fetchJson(url, withSomeDefaults)
-                results.push(...partialHistory.deployments)
-                offset = partialHistory.pagination.offset + partialHistory.pagination.limit
-                keepRetrievingHistory = partialHistory.pagination.moreData
+                try {
+                    const partialHistory: PartialDeploymentHistory<T> = await this.fetcher.fetchJson(url, withSomeDefaults)
+                    for (const deployment of partialHistory.deployments) {
+                        if (!foundIds.has(deployment.entityId)) {
+                            foundIds.add(deployment.entityId)
+                            yield deployment
+                        }
+                    }
+                    offset = partialHistory.pagination.offset + partialHistory.pagination.limit
+                    keepRetrievingHistory = partialHistory.pagination.moreData
+                } catch (error) {
+                    if (errorListener) {
+                        errorListener(`${error}`)
+                    }
+                    exit = true
```

Comment:
Are you stopping at the first error?

Reply:
Yes, we are stopping at the first error. We retry each fetch a few times, but if it fails, we don't continue fetching